tgui/packages/tgui/components/Dropdown.tsx

```diff
@@ -234,6 +237,55 @@ export class Dropdown extends Component<DropdownProps, DropdownState> {
     }
   }
 
+  getOptionValue(option): string {
+    return typeof option === 'string' ? option : option.value;
+  }
+
+  getOptionsValues(): string[] {
+    const { options = [] } = this.props;
+
+    return options.map((option) => {
+      return this.getOptionValue(option);
+    });
+  }
+
+  getSelectedIndex(): number {
+    const selected = this.state.selected || this.props.selected;
+    const { options = [] } = this.props;
+
+    return options.findIndex((option) => {
+      return this.getOptionValue(option) === selected;
+    });
+  }
+
+  toPrevious(): void {
+    const selectedIndex = this.getSelectedIndex();
+
+    if (selectedIndex < 0) {
+      return;
+    }
+
+    const options = this.getOptionsValues();
+    const endIndex = options.length - 1;
+    const previousIndex = selectedIndex === 0 ? endIndex : selectedIndex - 1;
+
+    this.setSelected(options[previousIndex]);
```

Comment:
Processing all the options for getting just one is a waste.
```suggestion
    const endIndex = this.props.options.length - 1;
    const previousIndex = selectedIndex === 0 ? endIndex : selectedIndex - 1;

    this.setSelected(this.getOptionValue(this.props.options[previousIndex]));
```

Reply:
`setSelected` accept string in argument, but this.props.options may be `string[] | DropdownEntry[]` in any case need get string from options

Reply:
That's true. However, instead of calling `getOptionsValues` it's enough to call `getOptionValue` once.

Reply:
True, fixed